src/main/java/net/snowflake/client/jdbc/SnowflakeStatementV1.java

```diff
@@ -217,7 +216,11 @@ long executeUpdateInternal(
       updateCount = ResultUtil.calculateUpdateCount(sfResultSet);
       queryID = sfResultSet.getQueryId();
       resultSetMetadataHandler(sfResultSet);
+    } catch (SnowflakeSQLException ex) {
+      queryID = ex.getQueryId();
```

Comment:
Will the query id always presence in response for any SQLException? If it's not avail, what will happen?

Reply:
This exception has a deafault `"unknown"` https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/jdbc/SnowflakeSQLException.java#L24 but of course it could be overriden by null when calling the constructor